server/client.go

```diff
@@ -2352,36 +2390,24 @@ func (c *client) addShadowSubscriptions(acc *Account, sub *subscription) error {
 			}
 			tokens = append(tokens, subj[start:])
 		}
-		if isSubsetMatch(tokens, im.prefix+im.from) {
-			ims = append(ims, im)
-		} else if hasWC {
-			if subjectIsSubsetMatch(im.prefix+im.from, subj) {
-				froms = append(froms, im)
-			}
+		if isSubsetMatch(tokens, im.to) {
+			ims = append(ims, ime{im, true})
+		} else if hasWC && subjectIsSubsetMatch(im.to, subj) {
+			ims = append(ims, ime{im, false})
 		}
 	}
 	acc.mu.RUnlock()
 
 	var shadow []*subscription
 
-	if len(ims) > 0 || len(froms) > 0 {
-		shadow = make([]*subscription, 0, len(ims)+len(froms))
+	if len(ims) > 0 {
+		shadow = make([]*subscription, 0, len(ims))
 	}
 
-	// Now walk through collected importMaps
-	for _, im := range ims {
+	// Now walk through collected stream imports that matched.
+	for _, ime := range ims {
```

Comment:
This is going to cause a copy of the `ime` struct. Is that intentional? If not, either convert ims to `[]*ime` or do a for loop with `i` so that you can do `ime := &ims[i]` to save a copy.

Reply:
Yes that was intentional, trying to avoid heap..

Reply:
Sure, but the `for range` is going to return a copy of `ime`. So it would be better to:
```
for i := 0; i < len(ims); i++{
   ime := &ims[i]
   ...
```

Reply:
True, was not worried about the copying for speed per se but was trying to avoid lots of garbage generation during subscriptions.

Reply:
Fixed.